src/Modules/UserProfileModule/components/AccountSetting/AccountSetting.tsx

```diff
@@ -0,0 +1,287 @@
+  userNameValidation,
+} from "../../../../utils/InputsValidation";
+import { baseUrl, getRequestHeaders, loader } from "../../../../utils/Utils";
+
+export default function AccountSetting() {
+  const { currentUser, getCurrentUser, saveAdminData } = useUser();
+  console.log(currentUser);
+  // instance from use navigate
+  const navigate = useNavigate();
+  // image input state
+  const [image, setImage] = useState(
+    `https://upskilling-egypt.com:3003/${currentUser?.imagePath}`
+  );
+  // hide pass or show state
+  let [hidePassInInpt, setHidePassInInpt] = useState(true);
+  // check loading state
+  const [isLoading, setIsLoading] = useState(false);
+  // function for change password input type
+  const changePassInputType = () => {
+    setHidePassInInpt((hidePassInInpt = !hidePassInInpt ? true : false));
+  };
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm({
+    criteriaMode: "all",
+  });
+
+  const getValues = () => {
+    return reset({
+      userName: currentUser?.userName,
+      email: currentUser?.email,
+      country: currentUser?.country,
+      phoneNumber: currentUser?.phoneNumber,
+      profileImage: image,
+    });
+  };
+  useEffect(() => {
+    getValues();
+  }, []);
+
+  const onSubmit = async (data: any) => {
+    setIsLoading(true);
+    const registerFormData = appendToFormData(data);
+    try {
+      const response = await axios.put(`${baseUrl}/Users/`, registerFormData, {
+        headers: getRequestHeaders(),
+      });
+      toast.success("Your new settings have been successfully saved");
+      getCurrentUser();
+      navigate("/dashboard/profile/account-info");
+      console.log(response.data.message);
+    } catch (err: any) {
+      toast.error(err.response.data.message);
+    }
+    setIsLoading(false);
+  };
+  const appendToFormData = (data: any) => {
+    const formData = new FormData();
+    formData.append("userName", data.userName);
+    formData.append("country", data.country);
+    formData.append("email", data.email);
+    formData.append("phoneNumber", data.phoneNumber);
+    formData.append("confirmPassword", data.confirmPassword);
+    formData.append("profileImage", data.profileImage[0]);
+    return formData;
+  };
+  return (
+    <>
+      <section>
+        <div className=" container pb-5">
+          <div className="form-body mt-3">
+            <form onSubmit={handleSubmit(onSubmit)}>
+              <div className="row justify-content-center ">
+                {/* image input */}
+                <div className="img-input">
+                  <div className="rounded-circle formImage mx-auto border">
+                    <input
+                      className={`inputImg w-100`}
+                      type="file"
+                      accept="image/*"
+                      placeholder="Recipe Price"
+                      {...register("profileImage")}
+                      onChange={({ target: { files } }) => {
```

Comment:
the register function adds an onChange handler. We can then get the values using getValues("profileImage")